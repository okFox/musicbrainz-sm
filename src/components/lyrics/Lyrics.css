
h3 {
    font-family: 'Acme', sans-serif;
}