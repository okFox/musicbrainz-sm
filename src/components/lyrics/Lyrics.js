import React from 'react';
import { useEffect, useState } from 'react';
import { getLyrics } from '../services/getLyrics';
import PropTypes from 'prop-types';
import './Lyrics.css';

function Lyrics({ match }) {
  const [lyrics, setLyrics] = useState('');
  const [lyricSong, setLyricSong] = useState('');
  const [lyricPerformer, setLyricPerformer] = useState('');

  useEffect(() => {
    getLyrics(match.params.artistName, match.params.recordingTitle)
      .then(receivedLyrics => setLyrics(receivedLyrics))
      .then(setLyricPerformer(match.params.artistName))
      .then(setLyricSong(match.params.recordingTitle));
  });

  return (
    <section>
      <h1>{lyricSong}</h1>
      <h3>by {lyricPerformer}</h3>
      <pre>{lyrics}</pre> 
    </section>
  );
}

Lyrics.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      recordingTitle: PropTypes.string.isRequired,
      artistName: PropTypes.string.isRequired
    })
  }).isRequired
};


export default Lyrics;
