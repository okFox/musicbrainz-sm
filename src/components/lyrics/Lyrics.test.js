import React from 'react';
import { shallow } from 'enzyme';
import Lyrics from './Lyrics';

describe('Lyrics component', () => {
  it('renders Lyrics', () => {
    const wrapper = shallow(<Lyrics match={{ albumId: '1', albumTitle: 'title' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
