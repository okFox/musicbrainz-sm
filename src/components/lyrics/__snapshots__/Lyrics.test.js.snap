// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics component renders Lyrics 1`] = `
<section>
  <h1 />
  <h3>
    by 
  </h3>
  <pre />
</section>
`;
